### PR TITLE
remove init blocks from all admission plugins

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:
removes init blocks from all admission plugins

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -154,6 +154,9 @@ func CreateKubeAPIServer(kubeAPIServerConfig *master.Config, sharedInformers inf
 
 // CreateKubeAPIServerConfig creates all the resources for running the API server, but runs none of them
 func CreateKubeAPIServerConfig(s *options.ServerRunOptions) (*master.Config, informers.SharedInformerFactory, *kubeserver.InsecureServingInfo, error) {
+	// register all admission plugins
+	registerAllAdmissionPlugins(s.Admission.Plugins)
```
Comment (p0lyn0mial) [inline cmd/kube-apiserver/app/server.go]: alternatively we could register from within init block.

`init() { registerAllAdmissionPlugins(&kubeapiserveradmission.Plugins) }`

File: cmd/kube-apiserver/app/plugins.go [hunk 2]
```diff
@@ -21,30 +21,58 @@ package app
...
+	"k8s.io/kubernetes/plugin/pkg/admission/deny"
+	"k8s.io/kubernetes/plugin/pkg/admission/exec"
+	"k8s.io/kubernetes/plugin/pkg/admission/gc"
+	"k8s.io/kubernetes/plugin/pkg/admission/imagepolicy"
+	"k8s.io/kubernetes/plugin/pkg/admission/initialresources"
+	"k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+	"k8s.io/kubernetes/plugin/pkg/admission/namespace/autoprovision"
+	"k8s.io/kubernetes/plugin/pkg/admission/namespace/exists"
+	"k8s.io/kubernetes/plugin/pkg/admission/namespace/lifecycle"
+	noderestriction "k8s.io/kubernetes/plugin/pkg/admission/noderestriction"
```
Comment (p0lyn0mial) [inline cmd/kube-apiserver/app/plugins.go]: I would like to rename these pkg names: `noderestriction`, `podpreset`, `podsecuritypolicy`, `default` in a follow up.

current name is `admission`, this is in conflict with `k8s.io/apiserver/pkg/admission`

File: cmd/kube-apiserver/app/server.go (hunk 1 above)
Comment (p0lyn0mial) [inline cmd/kube-apiserver/app/server.go]: @deads2k: my plan is to expose similar method on `genericapiserver ` and pass `s.Admission.Plugins` - just a few line below.

Comment (deads2k) [inline cmd/kube-apiserver/app/server.go]: This implementation is exactly what I was thinking of.  Actively register these in the plug registry that we're going to use.

File: pkg/kubeapiserver/admission/registry.go [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-*/
-
-package admission
-
-import (
-	"k8s.io/apiserver/pkg/admission"
-)
-
-// Plugins is a global registry for admission plugins.
-var Plugins = admission.Plugins{}
```
Comment (deads2k) [inline pkg/kubeapiserver/admission/registry.go]: yay!

File: cmd/kube-apiserver/app/plugins.go (hunk 2 above)
Comment (deads2k) [inline cmd/kube-apiserver/app/plugins.go]: > I would like to rename these pkg names: noderestriction, podpreset, podsecuritypolicy, default in a follow up.

Yes

**Comment by deads2k** [comment]
Great to see this come together

/lgtm

**Comment by deads2k** [comment]
@k8s-bot ok to test

**Comment by deads2k** [comment]
The admission change is the last in sequence that began with @sttts's pull to create separate plugin registries.

**Comment by p0lyn0mial** [comment]
just a small update to fix failing test

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Pull request merged**